piptools/scripts/compile.py

```diff
@@ -164,6 +164,12 @@ def cli(
     """
     log.verbosity = verbose - quiet
 
+    # If `src-files` was not provided as input, but rather as config,
+    # it will be part of the click Context `ctx`.
+    # However, is `src_files` is specified, then we want to use that.
+    if not src_files and ctx.default_map and "src_files" in ctx.default_map:
```

Comment:
```suggestion
    if not src_files and "src_files" in ctx.default_map:
```

I think that is sure to exist, have you tested?